File: C++/0045. Parentheses.cpp
// we have defined the necessary header files here for this problem.
// If additional header files are needed in your program, please import here.

int main()
{  
  // please define the C++ input here. For example: int a,b; cin>>a>>b;;  
  // please finish the function body here.  
  // please define the C++ output here. For example:cout<<____<<endl; 

    string s;
    while(cin>>s){
        int l = 0;
        int r = 0;
        for(auto c:s){
            if(c=='('){
                l++;
            }
            else if(l>0){
                l--;
            }
            else{
                r++;
            }
        }
        cout<<l+r<<endl;
    }
    return 0;
}

File: Python/0045. Parentheses.py
# If you need to import additional packages or classes, please import here.

def func():
    # please define the python3 input here. 
    # For example: a,b = map(int, input().strip().split())
    # please finish the function body here.
    # please define the python3 output here. For example: print().
    while True:
        try:
            s = input()
            l = 0
            r = 0
            for c in s:
                if c == '(':
                    l += 1
                elif l > 0:
                    l -= 1
                else:
                    r += 1
            print(l+r)
            
        except EOFError:
            break

    
if __name__ == "__main__":
    func()
